### PR TITLE
{compiler}[GCCcore/11.3.0] Clang v15.0.5 w/ Python 3.10.4

### PR DESCRIPTION
**Pull request by ocaisa**
Requires https://github.com/easybuilders/easybuild-easyblocks/pull/2800

### CODE REVIEW

File: easybuild/easyconfigs/c/Clang/Clang-15.0.5-GCCcore-11.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+]
+dependencies = [
+    # since Clang is a compiler, binutils is a runtime dependency too
+    ('binutils', '2.38'),
+    ('hwloc', '2.7.1'),
+    ('libxml2', '2.9.13'),
+    ('ncurses', '6.3'),
+    ('GMP', '6.2.1'),
+    ('Z3', '4.10.2'),
+    ('Python', '3.10.4'),
```
Comment (ocaisa) [inline easybuild/easyconfigs/c/Clang/Clang-15.0.5-GCCcore-11.3.0.eb:40]: `libclang` has python bindings, but I guess this is really a very optional dependency. I leave it in for now.

**Comment by ocaisa** [comment]
Test report by @ocaisa
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2800
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
cnx3 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz, Python 3.6.8
See https://gist.github.com/df42a8d2a914042976f8e46135b0a16b for a full test report.

File: easybuild/easyconfigs/c/Clang/Clang-15.0.5-GCCcore-11.3.0.eb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+description = """C, C++, Objective-C compiler, based on LLVM.  Does not
+ include C++ standard library -- use libstdc++ from GCC."""
+
+# Clang also depends on libstdc++ during runtime, but this dependency is
+# already specified as the toolchain.
+toolchain = {'name': 'GCCcore', 'version': '11.3.0'}
+
+source_urls = ["https://github.com/llvm/llvm-project/releases/download/llvmorg-%(version)s"]
+sources = [
+    'llvm-project-%(version)s.src.tar.xz',
```
Comment (smoors) [inline easybuild/easyconfigs/c/Clang/Clang-15.0.5-GCCcore-11.3.0.eb:24]: missing checksum

**Comment by ocaisa** [comment]
Test report by @ocaisa
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2800
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
cnx1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, Python 3.6.8
See https://gist.github.com/a8f265666a617e84776ee6a0155ba47b for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2800
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
zen2-ubuntu-2204 - Linux Ubuntu 22.04.1 LTS (Jammy Jellyfish), x86_64, AMD EPYC 7452 32-Core Processor (zen2), Python 3.10.6
See https://gist.github.com/e274171bc8293b92e94646f58f9662e7 for a full test report.

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by ocaisa** [comment]
@boegelbot please test @ generoso CORE_CNT=16

File: easybuild/easyconfigs/c/Clang/Clang-15.0.5-GCCcore-11.3.0.eb (hunk 1 above)
Comment (lexming) [reply easybuild/easyconfigs/c/Clang/Clang-15.0.5-GCCcore-11.3.0.eb:41]: We already have `Clang-Python-bindings` for the those binding. So I suggest to keep `Clang` without the Python bindings.

Comment (ocaisa) [reply easybuild/easyconfigs/c/Clang/Clang-15.0.5-GCCcore-11.3.0.eb]: Removed them and left a comment in place

**Comment by ocaisa** [comment]
@boegelbot please test @ generoso CORE_CNT=16

**Comment by ocaisa** [comment]
@boegelbot please test @ jsc-zen2 CORE_CNT=16

**Comment by ocaisa** [comment]
@boegelbot please test @ jsc-zen2 CORE_CNT=16

**Comment by ocaisa** [comment]
@boegelbot please test @ generoso CORE_CNT=16

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
zen2-ubuntu-2210 - Linux Ubuntu 22.10 (Kinetic Kudu), x86_64, AMD EPYC 7452 32-Core Processor (zen2), Python 3.10.7
See https://gist.github.com/SebastianAchilles/418ca08192bc90fe9701789402d38ff3 for a full test report.

**Review by SebastianAchilles — APPROVED** [review]
lgtm

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bwd-rockylinux-91 - Linux Rocky Linux 9.1 (Blue Onyx), x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), 2 x NVIDIA NVIDIA GeForce GTX 1060 6GB, 525.85.12, Python 3.9.14
See https://gist.github.com/SebastianAchilles/28ced88c1776d5250e00608eabaf58cc for a full test report.

**Pull request merged**